crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1543,9 +1592,19 @@ impl CumulocityConverter {
                 return Ok(vec![]);
             }
         };
-        let ops_file = ops_dir.join(c8y_operation_name);
+
         create_directory_with_defaults(&*ops_dir)?;
-        create_file_with_defaults(ops_file, None)?;
+        if let Some((template_name, Some(operation_name))) = self
+            .operations
+            .match_template_with_operation_by_command_name(c8y_operation_name)
```

Comment:
It took a lot of time to understand why this function works. It seems that the argument (`c8y_operation_name`) is the c8y's operation name (e.g. `c8y_Command`), however, the function checks if the template's command name (e.g. `command`) equals to the c8y operation name.

```rust
    /// Return operation and template name if `command_name` match incoming command
    pub fn match_template_with_operation_by_command_name(
        &self,
        command: &str,
    ) -> Option<(String, Option<String>)> {
        self.templates
            .iter()
            .find(|x| {
                x.command_name()
                    .is_some_and(|command_name| command_name.eq(command))
            })
            .map(|template| (template.name.clone(), template.on_fragment()))
    }
```

But then why it works? I figured it out that the problem deeper, is in `try_convert_data_message()`.

```rust
        match &channel {
            ...
                Channel::CommandMetadata { operation } => {
                    ...
                    OperationType::Custom(c8y_op_name) => {
                        dbg!(&c8y_op_name);
                        self.register_custom_operation(&source, c8y_op_name)
                    }
                }
            ...
        }
```

`OperationType::Custom(c8y_op_name)` holds actually **command** name! Not **c8y_op_name**!
Indeed I added as above, and it prints out `command`.
```
[crates/extensions/c8y_mapper_ext/src/converter.rs:1361:25] c8y_op_name = "command"
```

This needs to be fixed. Here `c8y_op_name` should hold the right value. Then, unfortunately, the function `match_template_with_operation_by_command_name()` needs to be rewrite to check if `template.on_fragment()` is equals to the c8y operation.

Reply:
After discussion with @rina23q and @albinsuresh, the fix was implemented in 451bb9146abda49a6cbe646643bf0006349628db